File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int,int> mp;

        for(int i=0; i<nums.size(); i++){
            if(mp.find(nums[i]) != mp.end()){
                result.push_back(i);
                result.push_back(mp[nums[i]]);
                return result;
            }
            else mp[target - nums[i]] = i;
        }
        return result;
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        checkSum = {}
        for i in range(len(nums)):
            num = target - nums[i]
            if num in checkSum:
                return [i, checkSum.get(num)]
            checkSum[nums[i]] = i
        return [-1, -1]